core/kazoo_csv/src/kz_csv.erl

```diff
@@ -88,31 +95,104 @@ fold(CSV, Fun, Acc)
 %% @doc
 %% @end
 %%--------------------------------------------------------------------
--spec take_row(csv()) -> {row(), csv()} |
-                         eof.
-take_row(<<>>) -> eof;
+-spec take_row(binary()) -> {row(), binary()} |
+                            'eof'.
```

Comment:
Why be so restrictive though. If someone leaves an empty line at the end of the file, you want to punish the request?